src/app/pages/storage2/modules/devices/components/zfs-info-card/zfs-info-card.component.ts

```diff
@@ -0,0 +1,178 @@
+import { DiskFormComponent } from 'app/pages/storage/disks/disk-form/disk-form.component';
+import { DevicesStore } from 'app/pages/storage2/modules/devices/stores/devices-store.service';
+import { WebSocketService, DialogService } from 'app/services';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-zfs-info-card',
+  templateUrl: './zfs-info-card.component.html',
+  styleUrls: ['./zfs-info-card.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ZfsInfoCardComponent implements OnInit {
+  @Input() topologyItem: VDev;
+  @Input() topologyParentItem: VDev;
+  @Input() disk: Disk;
+  private poolId: number;
+
+  get readErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.read_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.read_errors;
+  }
+
+  get writeErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.write_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.write_errors;
+  }
+
+  get checksumErrors(): number {
+    if (this.isMirror) {
+      return this.topologyItem.children.reduce((errors, vdev) => {
+        return errors + (vdev.stats?.checksum_errors || 0);
+      }, 0);
+    }
+    return this.topologyItem.stats.checksum_errors;
+  }
+
+  get isMirror(): boolean {
+    return this.topologyItem.type === VDevType.Mirror;
+  }
+
+  get isDisk(): boolean {
+    return this.topologyItem.type === VDevType.Disk;
+  }
+
+  get isOnline(): boolean {
+    return this.topologyItem.status === VDevStatus.Online;
+  }
+
+  constructor(
+    private route: ActivatedRoute,
+    private ws: WebSocketService,
+    private slideInService: IxSlideInService,
+    private dialogService: DialogService,
+    private matDialog: MatDialog,
+    private translate: TranslateService,
+    private devicesStore: DevicesStore,
+  ) {}
+
+  ngOnInit(): void {
+    this.poolId = +this.route.snapshot.paramMap.get('poolId');
+    this.slideInService.onClose$?.pipe(untilDestroyed(this)).subscribe(() => {
+      this.devicesStore.reloadList();
+    });
+  }
+
+  onEdit(): void {
+    const editForm = this.slideInService.open(DiskFormComponent, { wide: true });
+    editForm.setFormDisk(this.disk);
+  }
+
+  onOffline(): void {
+    this.dialogService.confirm({
+      title: this.translate.instant('Offline Disk'),
+      message: this.translate.instant('Offline disk {name}?', { name: this.disk.devname }),
+      buttonMsg: this.translate.instant('Offline'),
+    }).pipe(
+      filter(Boolean),
+      untilDestroyed(this),
+    ).subscribe(() => {
+      this.ws.call('pool.offline', [this.poolId, { label: this.topologyItem.guid }]).pipe(untilDestroyed(this)).subscribe(
+        () => {
+          this.devicesStore.reloadList();
```

Comment:
These subscribe and error call backs are same in `onOnline`, `onOffline` and `onDetach`. Let's create named functions vars and use them as call backs to avoid redundancy.

Reply:
Could you show an example?

Reply:
It wont work, keep as is